doc/devel/development_workflow.rst

```diff
@@ -452,48 +452,79 @@ Automated tests
-    - On the overview page *artifacts* are listed in the section *Related*.
-
-
-* Codecov and CodeQL are currently for information only. Their failure is not
-  necessarily a blocker.
-
 * tox_ is not used in the automated testing. It is supported for testing
   locally.
 
   .. _tox: https://tox.readthedocs.io/
 
-* If you know only a subset of CIs need to be run, this can be controlled on
-  individual commits by including the following substrings in commit messages:
-
-  - ``[ci doc]``: restrict the CI to documentation checks. For when you only
-    changed documentation (this skip is automatic if the changes are only under
-    ``doc/`` or ``galleries/``).
-  - ``[skip circle]``: skip the documentation build check. For when you didn't
-    change documentation.
-  - Unit tests can be turned off for individual platforms with
-
-    - ``[skip actions]``: GitHub Actions
-    - ``[skip appveyor]`` (must be in the first line of the commit): AppVeyor
-    - ``[skip azp]``: Azure Pipelines
+* Codecov and CodeQL are currently for information only. Their failure is not
+  necessarily a blocker.
 
-  - ``[skip ci]``: skip all CIs.  Use this only if you know your changes do not
-    need to be tested at all, which is very rare.
+Make sure the Linting, GitHub Actions, AppVeyor, CircleCI, and Azure pipelines are
+passing before merging. All checks are listed at the bottom of the GitHub page of your
+pull request.
+
+Here are some tips for finding the cause of a test failure:
+
+.. list-table::
+    :stub-columns: 1
+    :widths: 20 15 65
+
+    * - Linting
+      - Code style
+      - Errors are displayed as annotations on the pull request diff.
+    * - | Mypy
+        | Stubtest
+      - Type hints
+      - Errors are displayed as annotations on the pull request diff.
+    * - CircleCI
+      - reStructuredText (rst)
+      - Search the CircleCI log for ``WARNING``.
+    * - | GitHub Actions
+        | AppVeyor
+        | Azure pipelines
+      - Unit tests
+      - | Search the log for ``FAILURES``. Subsequent section should contain information
+          on failed tests.
+        | On Azure, find the images as *artifacts* of the Azure job:
+        |  Click *Details* on the check on the GitHub PR page.
+        |  Click *View more details on Azure Pipelines* to go to Azure.
+        |  On the overview page *artifacts* are listed in the section *Related*.
+
+Skip CI checks
+--------------
+
+If you know only a subset of CI checks need to be run, you can skip any unneeded CI
+checks on individual commits by including the following substrings in commit messages:
+
+.. list-table::
+    :stub-columns: 1
+    :widths: 25 20 55
+
+    * - ``[ci doc]``
+      - Only run documentation checks.
+      - | For when you have only changed documentation.
+        | ``[ci doc]`` is applied automatically when the changes are only to files in
+          ``doc/**/`` or ``galleries/**/``
+    * - ``[skip circle]``
```

Comment:
Out of curiosity: Why is there an asymmetry between `ci doc` and `skip circle`? Would `skip doc` or `ci circle` work as well?

Reply:
Looks like it would be a one line change to make `skip circle` into `skip doc`
https://github.com/matplotlib/matplotlib/blob/ff0497cc042332313afd3d2e9ba8f949c3857896/.circleci/config.yml#L20